com.avaloq.tools.ddk.xtext/src/com/avaloq/tools/ddk/xtext/resource/persistence/ResourceLoadMode.java

```diff
@@ -119,21 +125,42 @@ static ResourceLoadMode get(final Resource resource) {
   ResourceLoadMode FULL = of(c -> Instruction.LOAD, "FULL");
 
   /**
-   * Load mode for loading a resource fully from binary storage except for the {@link Constituent#NODE_MODEL}, which will be {@link Instruction#PROXY proxied}.
+   * Load mode for loading a resource fully from binary storage except for the {@link Constituent#NODE_MODEL} and {@link Constituent#SOURCE}, which will be
+   * {@link Instruction#PROXY proxied}.
+   */
+  ResourceLoadMode PROXIED_NODE_MODEL = of(c -> c == Constituent.NODE_MODEL || c == Constituent.SOURCE ? Instruction.PROXY
+      : Instruction.LOAD, "PROXIED_NODE_MODEL");
+
+  /**
+   * Load mode for loading a resource fully from binary storage except for the {@link Constituent#ASSOCIATIONS}, {@link Constituent#NODE_MODEL}, and
+   * {@link Constituent#SOURCE}, which will all be {@link Instruction#PROXY proxied}.
    */
-  ResourceLoadMode PROXIED_NODE_MODEL = of(c -> c == Constituent.NODE_MODEL ? Instruction.PROXY : Instruction.LOAD, "PROXIED_NODE_MODEL");
+  ResourceLoadMode PROXIED_NODE_MODEL_AND_ASSOCIATIONS = of(c -> c == Constituent.NODE_MODEL || c == Constituent.SOURCE || c == Constituent.ASSOCIATIONS
+      ? Instruction.PROXY
+      : Instruction.LOAD, "PROXIED_NODE_MODEL_AND_ASSOCIATIONS");
 
   /**
    * Load mode for loading a resource from binary storage without any node model.
    */
-  ResourceLoadMode NO_NODE_MODEL = of(c -> c == Constituent.NODE_MODEL ? Instruction.SKIP : Instruction.LOAD, "NO_NODE_MODEL");
+  ResourceLoadMode NO_NODE_MODEL = of(c -> c == Constituent.NODE_MODEL || c == Constituent.SOURCE ? Instruction.SKIP : Instruction.LOAD, "NO_NODE_MODEL");
 
   /**
-   * Load mode for demand-loading a resource's node model from binary storage. This is intended to be used in conjunction with {@link #PROXIED_NODE_MODEL}.
+   * Load mode for demand-loading a resource's node model from binary storage. This is intended to be used in conjunction with {@link #PROXIED_NODE_MODEL} and
+   * {@link #PROXIED_NODE_MODEL_AND_ASSOCIATIONS}.
    *
    * @return load mode for demand-loading a resource's node model, never {@code null}
    */
-  ResourceLoadMode ONLY_NODE_MODEL = of(c -> c == Constituent.RESOURCE || c == Constituent.NODE_MODEL ? Instruction.LOAD : Instruction.SKIP, "ONLY_NODE_MODEL");
+  ResourceLoadMode ONLY_NODE_MODEL = of(c -> c == Constituent.RESOURCE || c == Constituent.NODE_MODEL || c == Constituent.SOURCE ? Instruction.LOAD

```

Comment:
didn't quite get why Constituent.RESOURCE is needed here. I thought we expect this part to loaded already

Reply:
The RESOURCE constituent is somewhat of a dummy and is used by the PARSE load mode to indicate that the binary resource should be ignored. Vice versa `DirectLinkingResourceStorageLoadable` requires that the load mode specifies the LOAD instruction for RESOURCE.

I am of course open to other suggestions.